lib/rules/rulePortOmission.ts

```diff
@@ -0,0 +1,37 @@
+import { DiagnosticSeverity } from "vscode-languageserver";
+import { OEntity, OFile, OInstantiation } from "../parser/objects";
+import { IRule, RuleBase } from "./rulesBase";
+
+export class RuleCodingStyle extends RuleBase implements IRule {
+  public static readonly ruleName = 'coding-style';
+  file: OFile;
+
+  check() {
+    if (this.settings.style.portOmission) {
+      this.checkPortOmission();
+    }
+  }
+  checkPortOmission() {
+    for (const instantiation of this.file.objectList) {
+      if (instantiation instanceof OInstantiation) {
+        if (instantiation.definitions.length === 1) { // In case of multiple definitions this check will not be run (maybe create separate rule)
+          const definition = instantiation.definitions[0];
+          if (definition instanceof OEntity) {
+            // Check for entity instantiations only for now
```

Comment:
Is it a problem to also do this for subprograms, configurations, etc.?

Reply:
Activated also for components and configurations. I think subprograms normally have less parameters, and its harder to miss connections. (Which this rule tries to prevent)